### ISSUE TITLE
`function-no-unknown`: fix `linear` `xywh` `light-dark` false positives

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#7226]
### What minimal example or steps are needed to reproduce the bug?

```css
:root {
  color-scheme: light dark;

  --text-color: light-dark(#777, #000);
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "function-no-unknown": true
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo

### See also

https://drafts.csswg.org/css-color-5/#light-dark
https://github.com/mdn/browser-compat-data/pull/20935

### Do you have a proposal to fix the bug?

1. <del>either update https://github.com/niksy/css-functions-list/blob/master/index.json</del>
2. or add an exception

**Comment by ybiquitous** [comment]
@Mouvedia Good catch. I think sending a PR to `css-functions-list` is better.

**Comment by niksy** [comment]
Since this is currently in draft stage and it’s not documented on MDN, css-functions-list doesn’t include it. I could probably add `mdn/browser-compat-data` as new data source, but `light-dark` will eventually be documented on MDN.

**Comment by ybiquitous** [comment]
@Mouvedia @niksy Thanks for addressing this issue.

Until a reflection to `mdn/browser-compat-data` and a new version of `css-functions-list` is published, we can just add an exception to the `functionsList` below:

https://github.com/stylelint/stylelint/blob/2a1fc6dbcfc1f245a4f75e83d8d8048c098614c4/lib/rules/function-no-unknown/index.js#L51-L58

**Comment by niksy** [comment]
After additional consideration, I’ve decided to add those functions. `-webkit-gradient` is added manually, and `light-dark` is now pulled from [mdn/browser-compat-data](https://github.com/mdn/browser-compat-data).

If you’re familiar with additional resources, I can add them so we don’t have to add exceptions manually.

**Comment by Mouvedia** [comment]
- [ ] update css-function-list dependency to 3.2.1
- [ ] remove [this line](https://github.com/stylelint/stylelint/blob/2a1fc6dbcfc1f245a4f75e83d8d8048c098614c4/lib/rules/function-no-unknown/index.js#L54) and only this line
i.e. the inner functions `color-stop`, `from`, `to` are still unsupported

Thanks @niksy

**Comment by niksy** [comment]
OK, will try to manually add those 3 also, so Stylelint doesn’t need to have special cases.

Can you give me more information on `to` and `from` functions? Is there any (legacy) spec mentioning them?

**Comment by Mouvedia** [comment]
> OK, will try to manually add those 3 also, so Stylelint doesn’t need to have special cases.

IMHO you can wait; a commit will do.
i.e. it doesn't warrant a new version unless it adds _new_ functions like 3.2.1 did

> Can you give me more information on `to` and `from` functions?

see https://github.com/stylelint/stylelint/issues/5960#issuecomment-1763381499

> Is there any (legacy) spec mentioning them?

Not really but adding `-webkit-gradient` without them is incomplete/absurd from an user perspective but it's not a big deal for Stylelint because we already have them listed as exceptions.
ref https://developer.apple.com/library/archive/documentation/InternetWeb/Conceptual/SafariVisualEffectsProgGuide/Gradients/Gradient.html#//apple_ref/doc/uid/TP40008032-CH10-SW34
